jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,217 @@
+ * <p>
+ * Both lexicographical BFS and maximum cardinality search run in $\mathcal{O}(|V| + |E|)$. Checking whether given order
+ * is the perfect elimination order via {@link ChordalityInspector#isPerfectEliminationOrder(List)} takes
+ * $\mathcal{O}(|V| + |E|)$ as well. So, overall time complexity of the method
+ * {@link ChordalityInspector#isChordal()} is $\mathcal{O}(|V| + |E|)$.
+ *
+ * @param <V> the graph vertex type.
+ * @param <E> the graph edge type.
+ * @author Timofey Chudakov
+ * @see LexBreadthFirstIterator
+ * @see MaximumCardinalityIterator
+ * @since March 2018
+ */
+public class ChordalityInspector<V, E> {
+    /**
+     * The inspected graph.
+     */
+    private Graph<V, E> graph;
+    /**
+     * Contains true if the graph is chordal, otherwise false. Is null before the first call to the
+     * {@link ChordalityInspector#isChordal()}.
+     */
+    private Boolean chordal = null;
+    /**
+     * Order produced by {@code orderIterator}.
+     */
+    private List<V> order;
+    /**
+     * Iterator used for producing perfect elimination order.
+     */
+    private GraphIterator<V, E> orderIterator;
+    /**
+     * Creates a chordality inspector for {@code graph}, which uses {@link MaximumCardinalityIterator}
+     * as a default iterator.
+     *
+     * @param graph the graph for which a chordality inspector to be created.
+     */
+    public ChordalityInspector(Graph<V, E> graph) {
+        this(graph, IterationOrder.MCS);
+    }
+
+    /**
+     * Creates a chordality inspector for {@code graph}, which uses as iterator defined by the second
+     * parameter as an internal iterator.
+     *
+     * @param graph          the graph for which a chordality inspector to be created.
+     * @param iterationOrder the constant, which defines iterator to be used by this {@code ChordalityInspector}.
+     */
+    public ChordalityInspector(Graph<V, E> graph, IterationOrder iterationOrder) {
+        this.graph = Objects.requireNonNull(graph);
+        if (graph.getType().isDirected()) {
+            this.graph = new AsUndirectedGraph<>(graph);
+        }
+        if (iterationOrder == IterationOrder.MCS) {
+            this.orderIterator = new MaximumCardinalityIterator<>(graph);
+        } else {
+            this.orderIterator = new LexBreadthFirstIterator<>(graph);
+        }
+    }
+
+    /**
+     * Checks whether the inspected graph is chordal.
+     *
+     * @return true if this graph is chordal, otherwise false.
+     */
+    public boolean isChordal() {
+        if (chordal == null) {
+            order = lazyComputeOrder();
+            chordal = isPerfectEliminationOrder(order);
+        }
+        return chordal;
+    }
+
+    /**
+     * Returns the computed vertex order. In the case the inspected graph chordal, returned order
+     * is a perfect elimination order.
+     *
+     * @return computed vertex order.
+     */
+    public List<V> getOrder() {
```

Comment:
Is there a more specific name we can give this, e.g. getEliminationOrder?

Reply:
It is tricky, there is only the notion of "perfect elimination order", but I haven't encountered the notion of "elimination order". Either we can use the name "elimination order" and explain details in the Javadoc, or we can use "getComputedOrder".

Reply:
getSearchOrder?

Reply:
Great